pkg/controller/permissions/consumerpermissions.go

```diff
@@ -0,0 +1,157 @@
+package permissions
+
+import (
+	"context"
+
+	"github.com/acorn-io/baaah/pkg/router"
+	v1 "github.com/acorn-io/runtime/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/runtime/pkg/config"
+	imagerules "github.com/acorn-io/runtime/pkg/imagerules"
+	"github.com/acorn-io/runtime/pkg/labels"
+	"github.com/acorn-io/runtime/pkg/profiles"
+	"github.com/acorn-io/runtime/pkg/ref"
+	apierror "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetConsumerPermissions returns the permissions for a given container augmented with permissions from
+// any services it depends on that expose consumer permissions
+func GetConsumerPermissions(ctx context.Context, c kclient.Client, appInstance *v1.AppInstance, containerName string, container v1.Container) (result v1.Permissions, _ error) {
+	result = v1.Permissions{
+		ServiceName: containerName,
+	}
+
+	for _, dep := range container.Dependencies {
+		// This shouldn't happen, but okay?
+		if dep.TargetName == "" {
+			continue
+		}
+
+		svc := &v1.ServiceInstance{}
+		if err := ref.Lookup(ctx, c, svc, appInstance.Status.Namespace, dep.TargetName); apierror.IsNotFound(err) {
+			// We can ignore missing deps because the normal dep ordering will ensure that this container
+			// can't be created/updated until its deps are
+			continue
+		} else if err != nil {
+			return result, err
+		}
+
+		if svc.Spec.Consumer == nil {
+			continue
+		}
+
+		if svc.Spec.Consumer.Permissions != nil {
+			result.Rules = append(result.Rules, svc.Spec.Consumer.Permissions.GetRules()...)
+		}
+	}
+
+	return
+}
+
+func collectConsumerPermissions(ctx context.Context, client kclient.Client, app *v1.AppInstance) ([]v1.Permissions, error) {
+	result := []v1.Permissions{}
+
+	for containerName, containerDef := range app.Status.AppSpec.Containers {
+		consumerPerms, err := GetConsumerPermissions(ctx, client, app, containerName, containerDef)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, consumerPerms)
+	}
+
+	for jobName, jobDef := range app.Status.AppSpec.Jobs {
+		consumerPerms, err := GetConsumerPermissions(ctx, client, app, jobName, jobDef)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, consumerPerms)
+	}
+
+	return result, nil
+}
```

Comment:
Have you verified that sidecars are also covered by this? It is possible that the Dependencies field on a container object will include the deps of its sidecars, but idk for sure whether this is the case.

Reply:
Good point!
I was checking sidecars explicitly before, but they may have been dropped during another refactoring...
I will double check this and add a test case :+1:

Reply:
EDIT: Actually there's no change to the old logic here. We always only considered `app.Status.AppSpec.Jobs|Containers` - still I'll verify that again.

Reply:
Checked again and apparently, `dependsOn` in the sidecar scope within the Acornfile doesn't have any effect :thinking:

Reply:
I remember a conversation where the fact that `dependsOn` in sidecars were ignored.